src/Lean/Meta/Basic.lean

```diff
@@ -883,6 +883,29 @@ def withLocalDecl (name : Name) (bi : BinderInfo) (type : Expr) (k : Expr → n
 def withLocalDeclD (name : Name) (type : Expr) (k : Expr → n α) : n α :=
   withLocalDecl name BinderInfo.default type k
 
+partial def withLocalDecls 
+  [Inhabited α]
```

Comment:
We use 4 spaces for indenting the header.

Reply:
done.